files/external_node_v2.rb

```diff
@@ -344,7 +346,7 @@ def watch_and_send_facts(parallel)
           result = enc(certname)
           cache(certname, result)
         end
-      rescue TimeoutError, SocketError, Errno::EHOSTUNREACH, Errno::ECONNREFUSED
+      rescue TimeoutError, SocketError, Errno::EHOSTUNREACH, Errno::ECONNREFUSED, NodeRetrievalError
```

Comment:
It might be best to allow Net::HTTPServiceUnavailable (from the bug report) to be raised here and to use the cache, rather than _all_ node retrieval errors (e.g. a 40* error should be a hard failure, not cache).